In [ ]:
### What is the Superstore Dataset?

The Superstore dataset contains information about customers, their demographics, purchasing behavior, and their response to marketing campaigns. It includes variables such as income, marital status, product-wise spending, and the number of children at home.

### Why is it used?

This dataset is useful for customer segmentation, identifying spending patterns, and analyzing which types of customers are more likely to respond to marketing efforts. The insights can help businesses optimize their campaign strategies and improve customer targeting.


In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

Matplotlib is building the font cache; this may take a moment.


In [3]:
df = pd.read_csv("C:/Users/malus/Downloads/superstore_data.csv")

In [5]:
df.head()                  # Preview first 5 rows
df.shape                   # Check rows and columns
df.columns                 # View column names
df.info()                  # Data types and non-null values
df.describe()              # Summary statistics for numerical features

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 22 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Id                   2240 non-null   int64  
 1   Year_Birth           2240 non-null   int64  
 2   Education            2240 non-null   object 
 3   Marital_Status       2240 non-null   object 
 4   Income               2216 non-null   float64
 5   Kidhome              2240 non-null   int64  
 6   Teenhome             2240 non-null   int64  
 7   Dt_Customer          2240 non-null   object 
 8   Recency              2240 non-null   int64  
 9   MntWines             2240 non-null   int64  
 10  MntFruits            2240 non-null   int64  
 11  MntMeatProducts      2240 non-null   int64  
 12  MntFishProducts      2240 non-null   int64  
 13  MntSweetProducts     2240 non-null   int64  
 14  MntGoldProds         2240 non-null   int64  
 15  NumDealsPurchases    2240 non-null   i

Id   Year_Birth         Income      Kidhome     Teenhome  \
count   2240.000000  2240.000000    2216.000000  2240.000000  2240.000000   
mean    5592.159821  1968.805804   52247.251354     0.444196     0.506250   
std     3246.662198    11.984069   25173.076661     0.538398     0.544538   
min        0.000000  1893.000000    1730.000000     0.000000     0.000000   
25%     2828.250000  1959.000000   35303.000000     0.000000     0.000000   
50%     5458.500000  1970.000000   51381.500000     0.000000     0.000000   
75%     8427.750000  1977.000000   68522.000000     1.000000     1.000000   
max    11191.000000  1996.000000  666666.000000     2.000000     2.000000   

           Recency     MntWines    MntFruits  MntMeatProducts  \
count  2240.000000  2240.000000  2240.000000      2240.000000   
mean     49.109375   303.935714    26.302232       166.950000   
std      28.962453   336.597393    39.773434       225.715373   
min       0.000000     0.000000     0.000000         0.000000   
25%      24.000000    23.750000     1.000000        16.000000   
50%      49.000000   173.500000     8.000000        67.000000   
75%      74.000000   504.250000    33.000000       232.000000   
max      99.000000  1493.000000   199.000000      1725.000000   

       MntFishProducts  MntSweetProducts  MntGoldProds  NumDealsPurchases  \
count      2240.000000       2240.000000   2240.000000        2240.000000   
mean         37.525446         27.062946     44.021875           2.325000   
std          54.628979         41.280498     52.167439           1.932238   
min           0.000000          0.000000      0.000000           0.000000   
25%           3.000000          1.000000      9.000000           1.000000   
50%          12.000000          8.000000     24.000000           2.000000   
75%          50.000000         33.000000     56.000000           3.000000   
max         259.000000        263.000000    362.000000          15.000000   

       NumWebPurchases  NumCatalogPurchases  NumStorePurchases  \
count      2240.000000          2240.000000        2240.000000   
mean          4.084821             2.662054           5.790179   
std           2.778714             2.923101           3.250958   
min           0.000000             0.000000           0.000000   
25%           2.000000             0.000000           3.000000   
50%           4.000000             2.000000           5.000000   
75%           6.000000             4.000000           8.000000   
max          27.000000            28.000000          13.000000   

       NumWebVisitsMonth     Response     Complain  
count        2240.000000  2240.000000  2240.000000  
mean            5.316518     0.149107     0.009375  
std             2.426645     0.356274     0.096391  
min             0.000000     0.000000     0.000000  
25%             3.000000     0.000000     0.000000  
50%             6.000000     0.000000     0.000000  
75%             7.000000     0.000000     0.000000  
max            20.000000     1.000000     1.000000

### Task 1: Problem Definition & Dataset Overview

- Loaded the Superstore marketing dataset.
- Dataset contains 2,240 rows and 22 columns.
- Features include customer demographics, product spending, marketing response, etc.

In [10]:
df.isnull().sum()

Id                      0
Year_Birth              0
Education               0
Marital_Status          0
Income                 24
Kidhome                 0
Teenhome                0
Dt_Customer             0
Recency                 0
MntWines                0
MntFruits               0
MntMeatProducts         0
MntFishProducts         0
MntSweetProducts        0
MntGoldProds            0
NumDealsPurchases       0
NumWebPurchases         0
NumCatalogPurchases     0
NumStorePurchases       0
NumWebVisitsMonth       0
Response                0
Complain                0
dtype: int64

In [12]:
df['Income'] = df['Income'].fillna(df['Income'].median())


In [14]:
df.duplicated().sum()
df.drop_duplicates(inplace=True)


In [16]:
df['Dt_Customer'] = pd.to_datetime(df['Dt_Customer'])


In [18]:
df['Total_Spend'] = df[['MntWines', 'MntFruits', 'MntMeatProducts',
                        'MntFishProducts', 'MntSweetProducts', 'MntGoldProds']].sum(axis=1)


In [20]:
df['Family_Size'] = df['Kidhome'] + df['Teenhome']


In [22]:
df['Customer_Tenure_Days'] = (pd.to_datetime("2025-07-27") - df['Dt_Customer']).dt.days


In [24]:
    df['Education'] = df['Education'].str.strip().str.lower()
df['Marital_Status'] = df['Marital_Status'].str.strip().str.lower()


In [26]:
df.info()
df.describe()
df.head()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 25 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   Id                    2240 non-null   int64         
 1   Year_Birth            2240 non-null   int64         
 2   Education             2240 non-null   object        
 3   Marital_Status        2240 non-null   object        
 4   Income                2240 non-null   float64       
 5   Kidhome               2240 non-null   int64         
 6   Teenhome              2240 non-null   int64         
 7   Dt_Customer           2240 non-null   datetime64[ns]
 8   Recency               2240 non-null   int64         
 9   MntWines              2240 non-null   int64         
 10  MntFruits             2240 non-null   int64         
 11  MntMeatProducts       2240 non-null   int64         
 12  MntFishProducts       2240 non-null   int64         
 13  MntSweetProducts  

Id  Year_Birth   Education Marital_Status   Income  Kidhome  Teenhome  \
0   1826        1970  graduation       divorced  84835.0        0         0   
1      1        1961  graduation         single  57091.0        0         0   
2  10476        1958  graduation        married  67267.0        0         1   
3   1386        1967  graduation       together  32474.0        1         1   
4   5371        1989  graduation         single  21474.0        1         0   

  Dt_Customer  Recency  MntWines  ...  NumDealsPurchases  NumWebPurchases  \
0  2014-06-16        0       189  ...                  1                4   
1  2014-06-15        0       464  ...                  1                7   
2  2014-05-13        0       134  ...                  1                3   
3  2014-11-05        0        10  ...                  1                1   
4  2014-08-04        0         6  ...                  2                3   

   NumCatalogPurchases  NumStorePurchases  NumWebVisitsMonth  Response  \
0                    4                  6                  1         1   
1                    3                  7                  5         1   
2                    2                  5                  2         0   
3                    0                  2                  7         0   
4                    1                  2                  7         1   

   Complain  Total_Spend  Family_Size  Customer_Tenure_Days  
0         0         1190            0                  4059  
1         0          577            0                  4060  
2         0          251            1                  4093  
3         0           11            2                  3917  
4         0           91            1                  4010  

[5 rows x 25 columns]

In [28]:
df.to_csv("cleaned_superstore_data.csv", index=False)


In [31]:
# 1. Standardize Education column: capitalize first letter of each word
df['Education'] = df['Education'].str.strip().str.title()

# 2. Replace 'Phd' or 'PhD' with 'PHD'
df['Education'] = df['Education'].replace({'Phd': 'PHD', 'PhD': 'PHD'})

# 3. Optional: Capitalize first letter in Marital_Status
df['Marital_Status'] = df['Marital_Status'].str.strip().str.title()
